source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -330,20 +330,23 @@ def culling_journeys(resp, request):
     The candidate pool will be like [Journey_2, Journey_3]
     The sections set will be like set([Line 14, Line 6, Line 8, Bus 165])
     """
-    candidates_pool, sections_set = _build_candidate_pool_and_sections_set(resp)
+    candidates_pool, sections_set, idx_of_jrnys_must_keep = _build_candidate_pool_and_sections_set(resp)
 
-    nb_journeys_must_have = len(resp.journeys) - len(candidates_pool)
+    nb_journeys_must_have = len(idx_of_jrnys_must_keep)
     logger.debug("There are {0} journeys we must keep".format(nb_journeys_must_have))
-    nb_journeys_to_find = request["max_nb_journeys"] - nb_journeys_must_have
-    if nb_journeys_to_find <= 0:
-        # In this case, max_nb_journeys is smaller than nb_journeys_must_have, we have to make choices...
-        for jrny in candidates_pool:
-             journey_filter.mark_as_dead(jrny, 'Filtered by max_nb_journeys')
+    if request["max_nb_journeys"] == nb_journeys_must_have:
```

Comment:
this is gonna do nothing when you should instead just keep the must_have journeys